tests/kernel/stack/stack_api/src/test_stack_contexts.c

```diff
@@ -133,6 +133,40 @@ void test_stack_thread2isr(void)
 	tstack_thread_isr(&kstack);
 }
 
+/**
+ * @see k_stack_alloc_init(), k_stack_push(), #K_STACK_DEFINE(x), k_stack_pop(),
+ * k_stack_cleanup()
+ */
+void test_stack_alloc_thread2thread(void)
+{
+	int ret;
+
+#ifndef CONFIG_USERSPACE
+	k_thread_system_pool_assign(k_current_get());
```

Comment:
You shouldn't need to do this.
Just remove the ifdef around the call in test_main to the k_thread_resource_pool_assign. It works without userspace unless something is broken.